storage: move consistency checker to own file

tbg opened the pull request:
Pure code movement (mod running goimports). This is in preparation to
recomputing MVCCStats during consistency checks.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21201)
<!-- Reviewable:end -->

tbg committed:
storage: move consistency checker to own file

Pure code movement (mod running goimports). This is in preparation to
recomputing MVCCStats during consistency checks.

Release note: None

tschottdorf commented:
TFTR!
